src/Symfony/Bridge/Doctrine/ArgumentResolver/EntityValueResolver.php

```diff
@@ -0,0 +1,301 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\ArgumentResolver;
+
+use Doctrine\DBAL\Types\ConversionException;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\NoResultException;
+use Doctrine\Persistence\ManagerRegistry;
+use Doctrine\Persistence\ObjectManager;
+use Symfony\Bridge\Doctrine\Attribute\Entity;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Yields the entity matching the criteria provided in the route.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+final class EntityValueResolver implements ArgumentValueResolverInterface
+{
+    private array $defaultOptions;
+
+    public function __construct(
+        private ManagerRegistry $registry,
+        private ?ExpressionLanguage $language = null,
+        array $defaultOptions = []
+    ) {
+        $this->defaultOptions = array_merge([
+            'object_manager' => null,
+            'expr' => null,
+            'mapping' => [],
+            'exclude' => [],
+            'strip_null' => false,
+            'id' => null,
+            'evict_cache' => false,
+            'auto_mapping' => true,
+            'attribute_only' => false,
+        ], $defaultOptions);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(Request $request, ArgumentMetadata $argument): bool
+    {
+        if (0 === \count($this->registry->getManagerNames())) {
+            return false;
+        }
+
+        $options = $this->getOptions($argument);
+        if (null === $options['class']) {
+            return false;
+        }
+
+        if ($options['attribute_only'] && !$options['has_attribute']) {
+            return false;
+        }
+
+        // Doctrine Entity?
+        $objectManager = $this->getManager($options['object_manager'], $options['class']);
+        if (null === $objectManager) {
+            return false;
+        }
+
+        return !$objectManager->getMetadataFactory()->isTransient($options['class']);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resolve(Request $request, ArgumentMetadata $argument): iterable
+    {
+        $options = $this->getOptions($argument);
+
+        $name = $argument->getName();
+        $class = $options['class'];
+
+        $errorMessage = null;
+        if (null !== $options['expr']) {
```

Comment:
`if ($options['expr'] && !$object = $this->findViaExpression($class, $request, $options['expr'], $options)) {`

Reply:
I can not merge the 2 if, otherwith, it'll change the behavior when `null !== $options['expr']` and `null !== $object`.

The current implementation does nothing, but your suggestion will switch to the `else` block